packages/next/src/experimental/testmode/playwright/index.ts

```diff
@@ -1,29 +1,33 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
 import * as base from '@playwright/test'
 import type { NextFixture } from './next-fixture'
-import type { NextOptions } from './next-options'
+import type { NextOptions, NextOptionsConfig } from './next-options'
 import type { NextWorkerFixture } from './next-worker-fixture'
 import { applyNextWorkerFixture } from './next-worker-fixture'
 import { applyNextFixture } from './next-fixture'
+import { defaultPlaywrightConfig } from './default-config'
+
+export { defaultPlaywrightConfig }
 
 // eslint-disable-next-line import/no-extraneous-dependencies
 export * from '@playwright/test'
 
-export type { NextFixture, NextOptions }
-export type { FetchHandlerResult } from '../proxy'
-
-export interface NextOptionsConfig {
-  nextOptions?: NextOptions
+// Export this second so it overrides the one from `@playwright/test`
+export function defineConfig(
+  config: base.PlaywrightTestConfig<NextOptionsConfig>
+): base.PlaywrightTestConfig<NextOptionsConfig> {
+  return base.defineConfig<NextOptionsConfig>({
+    ...defaultPlaywrightConfig,
+    ...config,
+    use: {
+      ...defaultPlaywrightConfig.use,
+      ...config.use,
+    },
+  })
 }
 
-export function defineConfig<T extends NextOptionsConfig, W>(
```

Comment:
You do need all of these - that's how devs can create their own addons on top of our work.

Reply:
these are what are causing the type error